docs/rfcs/0002-ecommerce.md

```diff
@@ -0,0 +1,176 @@
+## Title for RFC
+ecommerce in mitxpro
+
+
+#### Abstract
+
+In MicroMasters we made some decisions about ecommerce. This RFC looks
+at the decisions made in MicroMasters and those made for edX's ecommerce
+stack and makes recommendations for ecommerce in mitxpro.
+
+#### Payment Processors
+
+The only supported payment processor for MicroMasters is CyberSource.
+At the time MIT didn't support other payment processors
+like Stripe for example, and I believe that's still true today.
+
+edX has support for multiple payment processors, for example 
+the [CyberSource payment processor backend](https://github.com/edx/ecommerce/blob/master/ecommerce/extensions/payment/processors/cybersource.py#L369).
+This adds some complexity since there are significant differences
+between how they interact with the web application's frontend and backend.
+
+##### Recommendation
+
+Unless MIT will support other payment processors in the near future, we should
+design to use CyberSource similar to what we do in MicroMasters.
+
+#### Models
+
+MicroMasters only allows a user to purchase one course at a time so there
+is no concept of a cart or basket. In MicroMasters the `Order` object stores
+the information about the order and has a foreign key to `Line` to store
+information about multiple purchased items. However since the user can only purchase one course at
+once, there is only one `Line` per `Order` in MicroMasters.
+
+It's a requirement for mitxpro to support multiple items per order so we'll need a cart or basket
+for the user to keep track of their order. edX uses a `Basket` object for this purpose, which subclasses
+the `AbstractBasket` class from django-oscar. When a user clicks the button to verify a course, they
+are redirected to the `basket/add/?sku=SKU` URL which is handled by django-oscar to add the
+item to the basket.
+
+##### Recommendation
+
+Whether or not we want to use django-oscar, we will need models like these (using django-oscar names here):
+ - `Order` for an instance of a purchase
+ - `Basket` for a user's shopping cart. This should be connected to `Order`.
+ - `Line` for an item in an `Basket`
+ - `Product` to represent a purchasable item. In MicroMasters we skip this and 
```

Comment:
Product question: will users ever be able to purchase something other than a single course run as a line item in their order? E.g.: some 'package' of courses or even an entire program's worth of courses. That will affect the structure of this model

Reply:
Maybe @pdpinch can answer better but I think there aren't plans to right now. But we should have the model be extensible enough to handle other similar kinds of purchases.